server/src/main/java/org/apache/kafka/server/share/ShareSessionCache.java

```diff
@@ -41,10 +41,10 @@ public class ShareSessionCache {
     private long numPartitions = 0;
 
     // A map of session key to ShareSession.
-    private Map<ShareSessionKey, ShareSession> sessions = new HashMap<>();
+    private final Map<ShareSessionKey, ShareSession> sessions = new HashMap<>();
 
     // Maps last used times to sessions.
-    private TreeMap<LastUsedKey, ShareSession> lastUsed = new TreeMap<>();
+    private final TreeMap<LastUsedKey, ShareSession> lastUsed = new TreeMap<>();
 
     // Visible for testing
     public synchronized TreeMap<LastUsedKey, ShareSession> lastUsed() {
```

Comment:
This seems a bit odd. This method exposes the TreeMap for testing, which breaks encapsulation and surely risks synchronization problems if used for anything other than testing. So, is there any point in the method being declared synchronized? Accessing the TreeMap in the caller is not going to be synchronized.

Reply:
Yeah, that's true this is bit odd and I have removed the method from the class itself. I don't see any usage of this method yet and if required by tests in future then we should send a copy of tree map instead the map itself.